Feature/build guides (#25)
kasenvr/vircadia-docs-sphinx: docs/source/developer/installer/INSTALLER.md

# Creating an Installer

During generation, CMake should produce an `install` target and a `package` target.

> The `install` target will copy the High Fidelity targets and their dependencies to your `CMAKE_INSTALL_PREFIX`.  

Reviewer: ```suggestion
The `install` target will copy the Project Athena targets and their dependencies to your `CMAKE_INSTALL_PREFIX`.  
```

> This variable is set by the `project(hifi)` command in `CMakeLists.txt` to `C:/Program Files/hifi` and stored in `build/CMakeCache.txt`

Reviewer: Lots of "HiFi" stuff in here - out of date?

Author: This needs to be updated in the main repo. The file will automatically be pulled from there. https://github.com/kasenvr/project-athena/pull/242

1. [nsisSlideshow Plug-in for Nullsoft](<http://wiz0u.free.fr/prog/nsisSlideshow/>) - 1.7
   1.  Extract Zip
   1.  Copy `bin\nsisSlideshow.dll` to `C:\Program Files (x86)\NSIS\Plugins\x86-ansi\`
   1.  Copy `bin\nsisSlideshowW.dll` to `C:\Program Files (x86)\NSIS\Plugins\x86-unicode\`

1. [Nsisunz plug-in for Nullsoft](http://nsis.sourceforge.net/Nsisunz_plug-in)
   1.  Download both Zips and unzip
   1.  Copy `nsisunz\Release\nsisunz.dll` to `C:\Program Files (x86)\NSIS\Plugins\x86-ansi\`
   1.  Copy `NSISunzU\Plugin unicode\nsisunz.dll` to `C:\Program Files (x86)\NSIS\Plugins\x86-unicode\`

1. [ApplicationID plug-in for Nullsoft]() - 1.0
   1.  Download [`Pre-built DLLs`](<https://github.com/connectiblutz/NSIS-ApplicationID/releases/download/1.1/NSIS-ApplicationID.zip>)
   1.  Extract Zip
   1.  Copy `Release\ApplicationID.dll` to `C:\Program Files (x86)\NSIS\Plugins\x86-ansi\`
   1.  Copy `ReleaseUnicode\ApplicationID.dll` to `C:\Program Files (x86)\NSIS\Plugins\x86-unicode\`

1. [npm](<https://www.npmjs.com/get-npm>)
    1.  Install version 10.15.0 LTS
    
1.  Perform a clean cmake from a new terminal.

> 1.  Open the `athena.sln` Solution and select the Release configuration.

Reviewer: ```suggestion
1.  Open the `athena.sln` solution and select the Release configuration.
```

> 1.  Build the Solution.

Reviewer: ```suggestion
1.  Build the solution.
```
